api/app/Http/Controllers/UserController.php

```diff
@@ -106,6 +107,33 @@ public function putOne($id, Request $request)
     }
 
     /**
+     * Patch an entity.
+     *
+     * @param  string   $id
+     * @param  Request  $request
+     * @return Response
+     */
+    public function patchOne($id, Request $request)
+    {
+        $this->validateId($id);
+        $this->repository->validateEmailConfirmationToken($request);
+        $model = $this->repository->find($id);
+
+        // Check if the email is being changed, and initialize confirmation
+        $email = $request->get('email');
+        if ($email and $model->email != $email) {
```

Comment:
Please use `&&` instead of `and` as the `and`/`or` keywords have unexpected behaviours.